lib/text/vtt_text_parser.js

```diff
@@ -149,16 +157,121 @@ shaka.text.VttTextParser = class {
     return region;
   }
 
+  /**
+   * Parses a style block into a Cue object.
+   *
+   * @param {!Array.<string>} text
+   * @param {!Map.<string, shaka.text.Cue>} styles
+   * @private
+   */
+  static parseStyle_(text, styles) {
+    // Skip empty blocks.
+    if (text.length == 1 && !text[0]) {
+      return;
+    }
+
+    // Skip comment blocks.
+    if (/^NOTE($|[ \t])/.test(text[0])) {
+      return;
+    }
+
+    // Only style block are allowed.
+    if (text[0] != 'STYLE') {
+      return;
+    }
+
+    if (!text[1].includes('::cue')) {
+      return;
+    }
+    let styleSelector = 'global';
+    const selector = text[1].match(/\((.*)\)/);
+    if (selector) {
+      styleSelector = selector.pop();
+    }
+
+    // We start at 2 to avoid '::cue' and end earlier to avoid '}'
+    let propertyLines = text.slice(2, -1);
+    if (text[1].includes('}')) {
+      const payload = /\{(.*?)\}/.exec(text[1]);
```

Comment:
Would you please add a comment to explain the regular expression, and probably an example? It would be faster for others to understand what it does.

Reply:
Done!